File: CSharp/104. Maximum Depth of Binary Tree.cs
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int val=0, TreeNode left=null, TreeNode right=null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
public class Solution {
    
    public int MaxDepth(TreeNode root) {
        if(root == null){
            return 0;
        }
        
        int ret1  = 1, ret2 = 1;
        
        if(root.left != null){
            ret1 = 1 + MaxDepth(root.left);
        }
        if(root.right != null){
            ret2 = 1 + MaxDepth(root.right);
        }
        
        return Math.Max(ret1, ret2);
        
    }
}

File: CPP/104. Maximum Depth of Binary Tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    int maxDepth(TreeNode* root) {
        if(root == NULL){
            return 0;
        }
        
        int ret1 = 0, ret2 = 0;
        if(root -> left != NULL){
            ret1 = 1 + maxDepth(root -> left);
        }
        if(root -> right != NULL){
            ret2 = 1 + maxDepth(root -> right);
        }
        
        return max(ret1 , max(ret2, 1));
    }
};